llvm/lib/Transforms/Utils/Local.cpp

```diff
@@ -3594,7 +3594,8 @@ DIExpression *llvm::getExpressionForConstant(DIBuilder &DIB, const Constant &C,
     return createIntegerExpression(C);
 
   auto *FP = dyn_cast<ConstantFP>(&C);
-  if (FP && (Ty.isFloatTy() || Ty.isDoubleTy())) {
+  if (FP &&
+      (Ty.isFloatTy() || Ty.isDoubleTy() || Ty.isHalfTy() || Ty.isBFloatTy())) {
```

Comment:
Can you use `isFloatingPointTy()` here to cover *all* FP types?

Reply:
Uploaded a patch to support: `fp128`, `x86_fp80`, `ppc_fp128`.

Reply:
@nikic Very sorry for my late response.
> Edit: Well, looking closer, this is bitcasting the float to an integer.

Looking at the documentation for `bitcastToAPInt`
```
// This function creates an APInt that is just a bit map of the floating
// point constant as it would appear in memory.  It is not a conversion,
// and treating the result as a normal integer is unlikely to be useful.
APInt IEEEFloat::bitcastToAPInt() const {
```
The function does not convert the `float` into an `int`.

Reply:
The FP value is being bitcast to an int value, so that it can appear in a DWARF expression. To the question about how debug info handles it, my understanding at the moment is that it's up to the consumer. I don't know specifically how different float types of the same size are handled in most debuggers, but there's not much more we can do from this side than provide a bit value and the variable's type AFAICT.

Reply:
@SLTozer Thanks for your additional input on DWARF expression.